integration-tests/tests/websocket/websockets.spec.js

```diff
@@ -14,40 +14,122 @@ See the License for the specific language governing permissions and
+    const wsConnectionSpan = await browser.globals.findSpan(span => span.name === 'connect');     
+    await browser.assert.ok(!!wsConnectionSpan, 'Connect span missing');     
+    await browser.click('#sendWs');
+    const wsMessage = await browser.globals.findSpan(span => span.name === 'onmessage');
+    await browser.assert.ok(!!wsMessage);
+    await browser.assert.strictEqual(wsMessage.kind, 'CONSUMER');
+    await browser.assert.strictEqual(wsMessage.tags['component'], 'websocket');
+    await browser.assert.strictEqual(wsMessage.tags['protocol'], '');
+    await browser.assert.strictEqual(wsMessage.tags['http.url'], browser.globals.wsBase);
+    await browser.assert.strictEqual(wsMessage.tags['http.response_content_length'], '7');
+
+    const wsSend = browser.globals.receivedSpans.find(span => span.name === 'send');
+    await browser.assert.ok(!!wsSend);
+    await browser.assert.strictEqual(wsSend.kind, 'PRODUCER');
+    await browser.assert.strictEqual(wsSend.tags['component'], 'websocket');
+    await browser.assert.strictEqual(wsSend.tags['protocol'], '');
+    await browser.assert.strictEqual(wsSend.tags['http.url'], browser.globals.wsBase);
+    await browser.assert.strictEqual(wsSend.tags['http.request_content_length'], '12');
+  },
+  'websocket constructor errors are captured': async function(browser) {
+    if (isSafari(browser)) {
+      return;
+    }
+    await browser.url(browser.globals.getUrl('/websocket/websocket-construct-errors.ejs'));
+    await browser.click('#connectWs');
+    const wsConnectionSpan = await browser.globals.findSpan(span => span.name === 'connect', -5000);     
+    await browser.assert.ok(!!wsConnectionSpan, 'Connect span missing');
+    await browser.assert.strictEqual(wsConnectionSpan.tags['error'], 'true');
+  },
+  'websocket send errors are captured': async function(browser) {
+    if (isSafari(browser)) {
+      return;
+    }
+    await browser.url(browser.globals.getUrl('/websocket/websocket-send-errors.ejs'));
+    await browser.click('#connectWs');
+    // not many ways to generate these errors, 
+    // trying to send msg during connect is the one I know but can be flaky
+    const wsSend = await browser.globals.findSpan(span => span.name === 'send');     
+    await browser.assert.ok(!!wsSend, 'Send span missing');
+    await browser.assert.strictEqual(wsSend.tags['error'], 'true');
+    await browser.assert.ok(wsSend.tags['error.message']);
+    await browser.assert.ok(wsSend.tags['error.object']);
+  },
+  'specifying sub-protocols does not break anything': async function(browser) {
+    if (isSafari(browser)) {
+      return;
+    }
+    const url = browser.globals.getUrl('/websocket/websocket-sub-protocol.ejs');
+    await browser.url(url);
+    await browser.click('#connectWs');
+    // Apparently there are leftover spans from last test even after clear
+    // as the spans arrive after clearing TODO fix somehow
```

Comment:
I'd suspect this is because even after the previous test finishes, the browser window is still open and some spans keep being sent. You can fix this by splitting scenarios into separate test files, this will result in separate collector servers.